pymtl3/passes/backends/verilog/tbgen/VerilogTBGenPass.py

```diff
@@ -21,19 +22,28 @@
 class VerilogTBGenPass( BasePass ):
   """ We only generate TB if it is imported or translation-imported """
 
+  # TBGenPass pass public pass data
+
+  #: tbgen case name
+  #:
+  #: Type: ``str``; input
+  #:
+  #: Default value: ""
+  case_name = MetadataKey(str)
+
   def __call__( self, top ):
     if not top._dsl.constructed:
       raise VerilogImportError( top,
         f"please elaborate design {top} before applying the TBGen pass!" )
 
-    top._tbgen = PassMetadata()
-    top._tbgen.tbgen_hooks = []
+    top._vtbgen = PassMetadata()
```

Comment:
Why not use the metadata APIs on `top` instead of having an attribute here? Ideally the way different passes communicate is through the metadata APIs?

Reply:
I forgot to do that. Do you have an example?

Reply:
For example, the import pass needs to know the file name of the translated result. The translation pass declares that as a metadata key here: https://github.com/cornell-brg/pymtl3/blob/b8debb763c46e96514642d1a4b45ea13c6f19522/pymtl3/passes/backends/verilog/translation/VerilogTranslationPass.py#L100
The ImportConfigs uses that metadata here: https://github.com/cornell-brg/pymtl3/blob/b8debb763c46e96514642d1a4b45ea13c6f19522/pymtl3/passes/backends/verilog/import_/VerilogVerilatorImportConfigs.py#L200

Reply:
I guess I need to modify every top._xxx added by those simulation passes and such